src/testUtils/builders.ts

```diff
@@ -9,62 +26,258 @@ import {
 export const randomDhis2Id = () =>
     faker.helpers.fromRegExp(/[a-zA-Z]{1}[a-zA-Z0-9]{10}/)
 
-export const testPager = ({
-    page = faker.number.int({ min: 0, max: 5 }),
-    total = faker.number.int({ min: 0, max: 50 }),
-    pageSize = 50,
-    pageCount = faker.number.int({ min: 0, max: 5 }),
+function randomValueIn<T>(list: T[]) {
+    return list[faker.number.int({ min: 0, max: list.length - 1 })]
+}
+
+const mockeryMapper = (keyName: string) => {
+    if (keyName === 'code') {
+        return () => faker.string.alphanumeric(6)
+    }
+    if (keyName === 'id') {
+        return () => randomDhis2Id()
+    }
+    return undefined
+}
+
+const { identifiable, referenceCollection, withAttributeValues } =
+    modelFormSchemas
+
+const UserSchema = identifiable.extend({
+    code: z.string().optional(),
+    displayName: z.string(),
+    username: z.string(),
+})
+
+const UserGroupSchema = identifiable.extend({
+    displayName: z.string(),
+})
+
+const AccessSchema = z.object({
+    delete: z.boolean(),
+    externalize: z.boolean(),
+    manage: z.boolean(),
+    read: z.boolean(),
+    update: z.boolean(),
+    write: z.boolean(),
+    data: z.object({
+        read: z.boolean(),
+        write: z.boolean(),
+    }),
+})
+
+const withDefaultListColumns = z.object({
+    displayName: z.string(),
+    lastUpdated: z.coerce.date(),
+    createdBy: UserSchema,
+    lastUpdatedBy: UserSchema,
+    access: AccessSchema,
+    href: z.string().url(),
+    sharing: z.object({ public: z.literal('rw------') }),
+})
+
+export const testAccess = (overwrites: Record<any, any> = {}) => ({
+    ...generateMock(AccessSchema, { mockeryMapper }),
+    ...overwrites,
+})
+
+export const testUser = (overwrites: Record<any, any> = {}) => ({
+    ...generateMock(UserSchema, { mockeryMapper }),
+    ...overwrites,
+})
+
+export const testUserGroup = (overwrites: Record<any, any> = {}) => ({
+    ...generateMock(UserGroupSchema, { mockeryMapper }),
+    ...overwrites,
+})
+
+const readOrganisationUnitColumns = z.object({
+    ancestors: z.array(organisationUnitSchema),
+    level: z.number().or(z.null()),
+    childCount: z.number(),
+})
+
+export const testIndicatorType = (overwrites: Record<any, any> = {}) => ({
+    ...generateMock(
+        IndicatorTypeSchema.merge(withDefaultListColumns).extend({
+            code: z.string(),
+        }),
+        { mockeryMapper }
+    ),
+    ...overwrites,
+})
+
+export const testIndicator = (overwrites: Record<any, any> = {}) => ({
+    ...generateMock(
+        identifiable.merge(withDefaultListColumns).extend({ code: z.string() }),
```

Comment:
maybe it would be better to add a placeholder for the Indicator schema so that we don't forget to update this?